### PR TITLE
docs: externalsPresets option

### LINKED ISSUE #3965: Document webpack change: move default externals from target into externalsPresets option
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/11447

---

**What kind of change does this PR introduce?**
refactoring

**Did you add tests for your changes?**
existing tests

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
* new option `externalsPresets` which allows to enable the default externals for various targets. Each is a boolean property in this object:
  * node
  * nodeWebkit (deprecated)
  * electronPreload
  * electronMain
  * web
  * webAsync

### PR DESCRIPTION
**Pull request by pranshuchittora**
Closes #3965

### CODE REVIEW

**Comment by anshumanv** [comment]
@pranshuchittora I started working on it https://github.com/webpack/webpack.js.org/issues/3965#issuecomment-691865456, no worries though, be mindful from next time :)

**Comment by pranshuchittora** [comment]
Pardon, didn't saw that.

File: src/content/configuration/externals.md [hunk 1]
```diff
@@ -401,6 +402,43 @@ console.log(head([1, 2, 3])); // logs 1 here
...
 
 T> Options like `output.chunkLoadTimeout`, `output.crossOriginLoading` and `output.scriptType` will also have effect on the external scripts loaded this way.
+
+## `externalsPresets`
+
+`object`
+
+Enable presets of externals for specific targets.
+
+W> Previously the following functionality was be achieved using `target`.
```
Comment (EugeneHlushko) [inline src/content/configuration/externals.md:415]:
```suggestion
W> In earlier webpack versions, the following functionality was achieved using [`target`](/configuration/target/).
```

File: src/content/configuration/externals.md [hunk 2]
```diff
@@ -401,6 +402,43 @@ console.log(head([1, 2, 3])); // logs 1 here
...
+`electronPreload`     | Treat electron built-in modules in the preload context like `web-frame`, `ipc-renderer` or `shell` as external and load them via require() when used.                     | boolean
+`electronRenderer`     | Treat electron built-in modules in the renderer context like `web-frame`, `ipc-renderer` or `shell` as external and load them via `require()` when used.                     | boolean
+`node`     | Treat node.js built-in modules like `fs`, `path` or `vm` as external and load them via `require()` when used.| boolean
+`nwjs`    | Treat `NW.js` legacy `nw.gui` module as external and load it via `require()` when used.| boolean
+`web`     | Treat references to `http(s)://...` and `std:...` as external and load them via `import` when used. __(Note that this changes execution order as externals are executed before any other code in the chunk)__.| boolean
+`webAsync`     | Treat references to 'http(s)://...' and 'std:...' as external and load them via async import() when used __(Note that this external type is an async module, which has various effects on the execution)__.| boolean
```
Comment (EugeneHlushko) [inline src/content/configuration/externals.md:428]:
```suggestion
`webAsync`     | Treat references to 'http(s)://...' and 'std:...' as external and load them via `async import()` when used __(Note that this external type is an `async` module, which has various effects on the execution)__.| boolean
```

File: src/content/configuration/externals.md [hunk 3]
```diff
@@ -401,6 +402,43 @@ console.log(head([1, 2, 3])); // logs 1 here
...
+
+
+__Example__
+
+Using `node` preset will not bundle built-in modules and treats them as external and loads them externally.
```
Comment (EugeneHlushko) [inline src/content/configuration/externals.md:433]:
```suggestion
Using `node` preset will not bundle built-in modules and treats them as external and loads them via `require()` when used.
```

File: src/content/configuration/externals.md [hunk 4]
```diff
@@ -401,6 +402,43 @@ console.log(head([1, 2, 3])); // logs 1 here
...
+
+__webpack.config.js__
+
+```js
+module.exports = {
+  // ...
+  externalsPresets:{
+    node:true
```
Comment (EugeneHlushko) [inline src/content/configuration/externals.md:441]:
```suggestion
    node: true
```

**Review by EugeneHlushko — CHANGES_REQUESTED** [review]

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**